velox/vector/DictionaryVector.h

```diff
@@ -152,7 +152,20 @@ class DictionaryVector : public SimpleVector<T> {
   }
 
   BaseVector* loadedVector() override {
-    dictionaryValues_ = BaseVector::loadedVectorShared(dictionaryValues_);
+    if (initialized_) {
+      return this;
+    }
+
+    SelectivityVector rows(dictionaryValues_->size(), false);
+    for (vector_size_t i = 0; i < this->size(); i++) {
+      if (!BaseVector::isNullAt(i)) {
+        auto ind = getDictionaryIndex(i);
+        rows.setValid(ind, true);
+      }
+    }
+    rows.updateBounds();
+
+    LazyVector::ensureLoadedRows(dictionaryValues_, rows);
```

Comment:
Should we do a VELOX_DCHECK that dictionaryValues_ is set after this call ?

Reply:
I can't see any code path would unset dictionaryValues_ as part of ensureLoadedRows (and we are checking dictionaryValues_ is valid at the constructor level). Should we check initialized_ is set instead?

Reply:
@kgpai It is not clear to me what would be the benefit of this check. Would you clarify?

Reply:
Since the expectation is that dictionaryValues_ is a shared pointer to the loaded base vector, my suggestion was just to check if that is indeed set and we dint suffer a slient failure. If it doesnt get unset then we can ignore.